### PR TITLE
Create Rotate.java

### PR DESCRIPTION
**Pull request by nitinrishab2000**
Function program to rotate 2-D array without using extra space.

### **Describe your change:**

- [ ] Add an algorithm?
- [ ] Fix a bug or typo in an existing algorithm?
- [ ] Documentation change?

### **Checklist:**

- [ ] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Java/blob/master/CONTRIBUTING.md).
- [ ] This pull request is all my own work -- I have not plagiarized.
- [ ] I know that pull requests will not be merged if they fail the automated tests.
- [ ] This PR only changes one algorithm file. To ease review, please open separate PRs for separate algorithms.
- [ ] All new Java files are placed inside an existing directory.
- [ ] All filenames are in all uppercase characters with no spaces or dashes.
- [ ] All functions and variable names follow Java naming conventions.
- [ ] All new algorithms have a URL in its comments that points to Wikipedia or other similar explanation.
- [ ] If this pull request resolves one or more open issues then the commit message contains `Fixes: #{$ISSUE_NO}`.

### CODE REVIEW

**Review by nitinrishab2000 — COMMENTED** [review]
Javadoc comments are added as per the format of code

File: Rotate.java [hunk 1]
```diff
@@ -0,0 +1,28 @@
+class Solution {
+    void rotateMatrix(int a[][], int N) {
```
[the remark this answers is not in the document]
Comment (AkMo3) [reply Rotate.java:2]: All right nice, commit the changes to the branch from where you are sending this PR to reflect the changes in actual file.

File: Rotate.java [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+        int n=N-1;
+        int start=0;
+        int end=n;
+        while(end>=start){
+            for(int k=start;k<end;k++){
+                int i=k;
+                int j=start;
+                int c=0;
+                int x=a[i][j];
+                int temp;
```
[the remark this answers is not in the document]
Comment (AkMo3) [reply Rotate.java:12]: Also place full stop as checkstyle sees it as Javadocs summary issue.
`*In one time total four rotation takes place`
`*these are four corners or second last from corner and so on.`

**Review by AkMo3 — APPROVED** [review]

**Comment by AkMo3** [comment]
Also since the code did not pass Code Formatter Check try running the following command in the folder where your java file is located.
1. `wget https://github.com/google/google-java-format/releases/download/google-java-format-1.9/google-java-format-1.9-all-deps.jar -O formatter.jar`
This will download a Google Java Formatter, exactly which used by this Check. Check .yml files in .github/workflows.
2. `java -jar formatter.jar --replace --set-exit-if-changed {FILE_NAME.java}`  This will format the code according to google standard format. 

Hope this solves the problem and your code gets merged. All the best.

**Comment by AkMo3** [comment]
You need to send the PR to **development** branch not **master** branch.

**Pull request closed without merging**